x-pack/plugins/alerting/server/rules_client/methods/create.ts

```diff
@@ -125,7 +124,15 @@ export async function create<Params extends RuleTypeParams = never>(
     params: updatedParams,
     actions,
   } = await withSpan({ name: 'extractReferences', type: 'rules' }, () =>
-    extractReferences(context, ruleType, data.actions, validatedAlertTypeParams)
+    extractReferences(
+      context,
+      ruleType,
+      (data.actions || []).map((action) => ({
+        ...action,
+        uuid: v4(),
```

Comment:
Actually using optional uuid on both update and create would make a lot of things easier, i just didn't want users to send duplicated UUIDs. But if we think those APIs should be the same, i can change it.

Reply:
Makes sense if we will not need a universally unique id.
If this is ok for the team i'll change it. 
WDYT @mikecote, @ymao1 and @doakalexi

Reply:
I think that makes sense to allow optionally specifying UUID on both create and update and to add a validation check to make sure any passed in UUIDs are unique to the rule. There is a small chance that a user could update a rule by removing an action with a UUID and adding a new action using the same UUID. This would cause their new action to be throttled using the values for the old action. But I think that probability is slim and being able to use the API for rules as code is a greater benefit.

Reply:
> There is a small chance that a user could update a rule by removing an action with a UUID and adding a new action using the same UUID.

That seems like something we could check.  We just _another_ UUID, which is hidden from the user completely.

**I'm kidding!**

We could definitely do some basic checks, like make sure if a UUID is reused between old and new actions, that the connector types are the same.   Guessing that may be all we could check with though, so you'd still have the problem Ying described if the actions being swapped were the same type.

If we decide to go this route, let's open an issue to see if we can fix it.

In terms of future breaking changes, seems safer to **NOT** accept the UUIDs today, then we can accept them in a future release.  Will be harder if we accept them now and decide that we don't want to in the future.

So, maybe just go with what we have, open an issue to discuss whether we should allow UUIDs on create?

Reply:
1- Too late, i've already changed it :)
2- Even if we kept it, the scenario would've still been possible since it happens trough the update API (it's still the same). And we have to allow update API to get uuid. 

I'll create an issue to investigate if we can fix it.